solidity.js

```diff
@@ -91,114 +236,15 @@ function hljsDefineSolidity(hljs) {
-    //(old-style, with an underscore)
-    var SOL_ASSEMBLY_MEMBERS_OLD = {
-        begin: /_/,
-        end: /[^A-Za-z0-9$.]/,
-        excludeBegin: true,
-        excludeEnd: true,
-        keywords: {
-            built_in: 'slot offset'
-        },
-        relevance: 2,
-    };
-
-    //covers the special slot/offset notation in assembly
-    //(new-style, with a dot; keeping this separate as it
-    //may be expanded in the future)
-    var SOL_ASSEMBLY_MEMBERS = {
-        begin: /\./,
-        end: /[^A-Za-z0-9$.]/,
-        excludeBegin: true,
-        excludeEnd: true,
-        keywords: {
-            built_in: 'slot offset length'
-        },
-        relevance: 2,
-    };
-
-    function isNegativeLookbehindAvailable() {
-        try {
-            new RegExp('(?<!.)');
-            return true;
-        } catch (_) {
-            return false;
-        }
-    }
-
-    //like a C number, except:
-    //1. no octal literals (leading zeroes disallowed)
-    //2. underscores (1 apiece) are allowed between consecutive digits
-    //(including hex digits)
-    //also, all instances of \b (word boundary) have been replaced with (?<!\$)\b
-    //NOTE: we use string rather than regexp in the case where negative lookbehind
-    //is allowed to avoid Firefox parse errors; sorry about the resulting double backslashes!
-    if (isNegativeLookbehindAvailable()) {
-        var SOL_NUMBER_RE = '-?((?<!\\$)\\b0[xX]([a-fA-F0-9]_?)*[a-fA-F0-9]|((?<!\\$)\\b[1-9](_?\\d)*(\\.((\\d_?)*\\d)?)?|\\.\\d(_?\\d)*)([eE][-+]?\\d(_?\\d)*)?|(?<!\\$)\\b0)';
-    } else {
-        var SOL_NUMBER_RE = /-?(\b0[xX]([a-fA-F0-9]_?)*[a-fA-F0-9]|(\b[1-9](_?\d)*(\.((\d_?)*\d)?)?|\.\d(_?\d)*)([eE][-+]?\d(_?\d)*)?|\b0)/;
-    }
-
-
-    var SOL_NUMBER = {
-        className: 'number',
-        begin: SOL_NUMBER_RE,
-        relevance: 0,
-    };
-
-    var HEX_APOS_STRING_MODE = {
-        className: 'string',
-        begin: /\bhex'(([0-9a-fA-F]{2}_?)*[0-9a-fA-F]{2})?'/, //please also update HEX_QUOTE_STRING_MODE
-    };
-    var HEX_QUOTE_STRING_MODE = {
-        className: 'string',
-        begin: /\bhex"(([0-9a-fA-F]{2}_?)*[0-9a-fA-F]{2})?"/, //please also update HEX_APOS_STRING_MODE
-    };
-
-    //I've set these up exactly like hljs's builtin STRING_MODEs,
-    //except with the optional initial "unicode" text
-    var SOL_APOS_STRING_MODE = hljs.inherit(hljs.APOS_STRING_MODE, //please also update SOL_QUOTE_STRING_MODE
-        { begin: /\b(unicode)?'/ }
-    );
-    var SOL_QUOTE_STRING_MODE = hljs.inherit(hljs.QUOTE_STRING_MODE, //please also update SOL_APOS_STRING_MODE
-        { begin: /\b(unicode)?"/ }
-    );
+    //NOTE: including "*" as a "lexeme" because we use it as a "keyword" below
+    var SOL_LEXEMES_RE = /[A-Za-z_$][A-Za-z_$0-9]*|\*/;
 
     var SOL_FUNC_PARAMS = {
         className: 'params',
         begin: /\(/, end: /\)/,
         excludeBegin: true,
         excludeEnd: true,
+        lexemes: SOL_LEXEMES_RE,
```

Comment:
As mentioned before this is going away with v11. Use `keywords.$pattern` instead.

Reply:
I think I'll just handle that on #41 once I figure out what's going wrong there and fix it...